tests/integrationv2/test_jdk_happy_path.py

```diff
@@ -0,0 +1,49 @@
+import copy
+import pytest
+
+from configuration import available_ports
+from common import ProviderOptions, Protocols, Certificates, data_bytes
+from fixtures import managed_process
+from providers import Provider, S2N, JavaSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version
+
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", [Certificates.ECDSA_384], ids=get_parameter_name)
+def test_s2n_server_happy_path(managed_process, certificate, protocol):
+
+    port = next(available_ports)
+    random_bytes = data_bytes(100)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        protocol=protocol,
+        data_to_send=random_bytes)
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(JavaSSL, client_options, timeout=5)
+
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+
+    expected_version = get_expected_s2n_version(protocol, S2N)
+
+    for results in server.get_results():
+        assert results.exception is None
+        ''' 
+        The JDK sends a user_canceled alert as well as a close_notify alert after 
```

Comment:
Does this existing happy path test work with the JDK? I know this is just testing stages, but I'd be interested to know how many existing tests work without modification.

I'm guessing that exit code will break everything, and we would need to ignore the exit code if JDK. Maybe there is a way to turn off alerts in the jdk client?

Reply:
Hmm I can look into turning off alerts. And if not, it would be easy to add a conditional to the exit code. The reason I created a separate happy path test for the jdk is because the s2n certificates are in the pem format, whereas the jdk takes certificates in the pkcs12 format. I used Openssl to convert a pem cert to a p12 cert and I don't think it would be worth it to do that for all the certs in our test directory. Would love any suggestions to get around this issue though.